### ISSUE TITLE
EvalError: List is not defined in .\src\List.js

### ISSUE THREAD

**Issue by GabbeV** [callstack/linaria#1153]
## Environment

- Linaria version: latest
- Bundler (+ version): webpack 5.75.0
- Node.js version: 18.12.1 / 16.14.2
- OS: windows 10 / stackblitz (linux?)

## Description

Linaria transforms the following code 

```JS
import { css } from '@linaria/core';
import { listColor } from './theme';

const style = css`
  background-color: ${listColor};
`;

export default function List({ children }) {
  return <ul className={style}>{children}</ul>;
}

List.Item = function ListItem({ children }) {
  return <li>{children}</li>;
};
```

into

```JS
"use strict";

var _theme = require("./theme");
const _exp = /*#__PURE__*/() => _theme.listColor;
List.Item = function ListItem() {
  return null;
};
exports.__linariaPreval = {
  _exp: _exp
};
```

which fails to evalutate with EvalError

```
ERROR in ./src/List.js
Module build failed (from ./node_modules/@linaria/webpack5-loader/lib/index.js):
EvalError: List is not defined in
| .\src\List.js

    at .\node_modules\@linaria\babel-preset\lib\module.js:342:15
    at Array.forEach (<anonymous>)
    at Module.evaluate (.\node_modules\@linaria\babel-preset\lib\module.js:317:10)
    at evaluate (.\node_modules\@linaria\babel-preset\lib\evaluators\index.js:20:5)
    at evalStage (.\node_modules\@linaria\babel-preset\lib\transform-stages\2-eval.js:26:45)
    at syncStages (.\node_modules\@linaria\babel-preset\lib\transform.js:46:45)
    at transform (.\node_modules\@linaria\babel-preset\lib\transform.js:146:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
 @ ./src/index.js 1:0-26 7:43-47 8:31-40 10:24-33 12:24-33 14:24-33
```

## Reproducible Demo

https://stackblitz.com/edit/linaria-bug-p7zcrl?file=src%2FList.js

**Comment by kabbi** [comment]
Possibly related: linaria shaker also removes class methods, including private ones, which may make the resulting code crash with SyntaxError on subsequent eval step by linaria

Method `createDom` is removed in the following code by shaker:
```
import { dragHandleStyle } from './styles';

export class BlockHandleDOM {
  readonly dom$: HTMLElement;

  constructor() {
    this.dom$ = this.#createDOM();
    this.dom$.classList.add(dragHandleStyle, 'hide');
  }

  #createDOM() {
    const dom = document.createElement('diva');
    return dom;
  }
}

```

Exception:
```
ERROR in ./src/components/App.tsx
Module build failed (from ../node_modules/@linaria/webpack-loader/lib/index.js):
SyntaxError: /Users/kabbi/dev/work/fibery/linaria-issue/app/src/components/block-handle-dom.ts: Private name #createDOM is not defined. (5:21)

  3 |   readonly dom$: HTMLElement;
  4 |   constructor() {
> 5 |     this.dom$ = this.#createDOM();
    |                      ^
  6 |     this.dom$.classList.add(dragHandleStyle, 'hide');
  7 |   }
```

**Comment by layershifter** [comment]
https://stackblitz.com/edit/linaria-bug-bjaanx

This have been fixed in v5, AFAIR in #1337. @kabbi your case have been also fixed 🎉
